case/interactive/T0000_rackhd_OSInstall.py

```diff
@@ -114,13 +117,21 @@ def test(self):
             return
 
         # Start workflow to install OS
+        node_workflow_dic = {}
         for node_id, node in node_os_install_list:
             node_bmc_ip = node.get_bmc_ip()
             self.monorail.put_node_ipmi_obm(
                 node_id=node_id, host=node_bmc_ip, username="admin", password="admin")
             node.install_os(self.data["os_name"])
             self.log(
                 'WARNING', 'Installation on node {} starts, please open VNC to check'.format(node_id))
+            rsp = node.get_workflows(active=True).items()
+            if len(rsp) != 0:
+                workflow_obj = rsp[0][1]
+                workflow_instance_id = workflow_obj.instanceId
+                node_workflow_dic[node_id] = workflow_instance_id
+            else:
+                self.log("WARNING", "OS installation workflow is not active on node ID: {}".format(node_id))
```

Comment:
I think is is a result FAIL condition

Reply:
Other nodes' installation should also continue even the specific node fails here, and fail result of other nodes may overwrite the last failure in junit. So we choose to report a summary of all failures in the end.

Reply:
`self.result(FAIL, msg)` will mark case as fail, and APPEND msg to the final summary, you don't have worry about overwrite.
`self.log("WARNING", msg)` here will print a warning message to log file and stdout, but nothing in final summary.

Reply:
OK, I see. Anyway, there is summary result for all the failed nodes of "OS installation workflow is not active" in the end now. So no worry for loosing record of installation workflow not active. Is that good?